### DISCUSSION TITLE
Groff/Ghostscript minor problem

### DISCUSSION THREAD

**Discussion by wssimms** [Homebrew/discussions#5084] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9c8374d3b78c4f531c5c2c0d272d1fb1f1b60d70
Last commit: 3 days ago
Core tap HEAD: 028c61a4955c3eb1bb9cf50801aad21d9cbce4ff
Core tap last commit: 2 days ago
Core tap JSON: 24 Jan 02:09 UTC
Core cask tap JSON: 24 Jan 02:09 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libpas2jslib.dylib
```

### Description of issue

I was trying to use groff to produce a PDF with embedded fonts (option -P-e for gropdf) and was getting the error:

`/opt/homebrew/bin/gropdf:test.roff: warning: The download file in '/opt/homebrew/Cellar/groff/1.23.0/share/groff/1.23.0/font/devpdf'  has erroneous entry for 'Times-Roman (TR)'`

It turned out the problem was that the file `/opt/homebrew/Cellar/groff/1.23.0/share/groff/1.23.0/font/devpdf/download` had pathnames like `*/opt/homebrew/Cellar/ghostscript/10.01.2/share/ghostscript/10.01.2/Resource/Font/NimbusRoman-Regular` although the ghostscript version is actually 10.02.1.

That is, 10.01.2 instead of the correct 10.02.1.

Correcting this allowed me to embed the fonts.

**Comment by cskeeters** [comment]
I also had the problem where fonts worked when groff outputs to ps and then ps2pdf converts to pdf, but `pdfmom` would output an error.  I came up with a manual solution until this gets fixed properly.

```bash
cd /opt/homebrew/share/ghostscript
ln -s 10.03.1 current
```

Then I edited `/opt/homebrew/share/groff/current/font/devpdf/download` (version non-specific) to be:

	AvantGarde-Book	/opt/homebrew/share/ghostscript/current/Resource/Font/URWGothic-Book
	AvantGarde-BookOblique	/opt/homebrew/share/ghostscript/current/Resource/Font/URWGothic-BookOblique
	AvantGarde-Demi	/opt/homebrew/share/ghostscript/current/Resource/Font/URWGothic-Demi
	AvantGarde-DemiOblique	/opt/homebrew/share/ghostscript/current/Resource/Font/URWGothic-DemiOblique
	Bookman-Demi	/opt/homebrew/share/ghostscript/current/Resource/Font/URWBookman-Demi
	Bookman-DemiItalic	/opt/homebrew/share/ghostscript/current/Resource/Font/URWBookman-DemiItalic
	Bookman-Light	/opt/homebrew/share/ghostscript/current/Resource/Font/URWBookman-Light
	Bookman-LightItalic	/opt/homebrew/share/ghostscript/current/Resource/Font/URWBookman-LightItalic
	Courier	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusMonoPS-Regular
	Courier-Bold	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusMonoPS-Bold
	Courier-BoldOblique	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusMonoPS-BoldItalic
	Courier-Oblique	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusMonoPS-Italic
	FreeEuro	../devps/freeeuro.pfa
	Helvetica	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusSans-Regular
	Helvetica-Bold	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusSans-Bold
	Helvetica-BoldOblique	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusSans-BoldItalic
	Helvetica-Narrow	/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusSansNarrow-Regular
	Helvetica-Narrow-Bold	/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusSansNarrow-Bold
	Helvetica-Narrow-BoldOblique	/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusSansNarrow-BoldOblique
	Helvetica-Narrow-Oblique	/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusSansNarrow-Oblique
	Helvetica-Oblique	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusSans-Italic
	NewCenturySchlbk-Bold	/opt/homebrew/share/ghostscript/current/Resource/Font/C059-Bold
	NewCenturySchlbk-BoldItalic	/opt/homebrew/share/ghostscript/current/Resource/Font/C059-BdIta
	NewCenturySchlbk-Italic	/opt/homebrew/share/ghostscript/current/Resource/Font/C059-Italic
	NewCenturySchlbk-Roman	/opt/homebrew/share/ghostscript/current/Resource/Font/C059-Roman
	Palatino-Bold	/opt/homebrew/share/ghostscript/current/Resource/Font/P052-Bold
	Palatino-BoldItalic	/opt/homebrew/share/ghostscript/current/Resource/Font/P052-BoldItalic
	Palatino-Italic	/opt/homebrew/share/ghostscript/current/Resource/Font/P052-Italic
	Palatino-Roman	/opt/homebrew/share/ghostscript/current/Resource/Font/P052-Roman
	Symbol	*/opt/homebrew/share/ghostscript/current/Resource/Font/StandardSymbolsPS
	Times-Bold	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusRoman-Bold
	Times-BoldItalic	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusRoman-BoldItalic
	Times-Italic	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusRoman-Italic
	Times-Roman	*/opt/homebrew/share/ghostscript/current/Resource/Font/NimbusRoman-Regular
	ZapfChancery-MediumItalic	/opt/homebrew/share/ghostscript/current/Resource/Font/Z003-MediumItalic
	ZapfDingbats	*/opt/homebrew/share/ghostscript/current/Resource/Font/D050000L

**Reply by SMillerDev** [reply]
Could you suggest a very simple test to reproduce this? That way we'll add it to the groff formula catch this issue next update
